execution/src/posix/execution.cc

```diff
@@ -139,8 +146,37 @@ namespace leatherman { namespace execution {
         return fs.st_mode & S_IXOTH;
     }
 
-    string which(string const& file, vector<string> const& directories)
+    bool isBuiltin(string const& file)
+    {
+        string data;
+        bool builtin;
+        try {
+            string cmd = "type ";
+            cmd.append(file);
+            const int max_buffer = 25 + file.length();
+            char buffer[max_buffer];
+            FILE *stream = popen(cmd.c_str(), "r");
+            if (stream) {
+                while (!feof(stream)) {
+                    if (fgets(buffer, max_buffer, stream) != NULL)
+                        data.append(buffer);
+                }
+                pclose(stream);
+            }
+            builtin = boost::algorithm::contains(data, "builtin");
+        } catch (exception& e) {
+            std::cout << e.what() << std::endl;
```

Comment:
can we be more specific about which part of the code can raise an exception? can it even raise an exception?